internal/controller/scheduler/pod_watcher.go

```diff
@@ -571,15 +586,13 @@ func (w *podWatcher) jobCancelChecker(ctx context.Context, stopCh <-chan struct{
 
 		switch job.State {
 		case api.JobStateCanceled, api.JobStateCanceling:
-			log.Info("Evicting pending pod for cancelled job")
-			eviction := &policyv1.Eviction{ObjectMeta: podMeta}
-			if err := w.k8s.PolicyV1().Evictions(w.cfg.Namespace).Evict(ctx, eviction); err != nil {
-				log.Error("Couldn't evict pod", zap.Error(err))
-				podEvictionErrorsCounter.WithLabelValues("bk_job_cancelled", string(kerrors.ReasonForError(err))).Inc()
+			log.Info("Deleting pending pod for cancelled job")
+			if err := w.forcefullyDeletePod(ctx, log, &podMeta, "job_cancelled"); err != nil {
+				// Low likelihood, we will retry.
 				continue
+			} else {
+				return
```

Comment:
The `else` here is redundant because of the `continue`. Consider:

```suggestion
			}
			return
```